src/smt/env_obj.h

```diff
@@ -44,6 +44,12 @@ class EnvObj
    */
   Node rewrite(TNode node);
 
+  /**
+   * Garbage collect the rewrite caches.
+   * This is a wrapper around theory::Rewriter::clearCaches via Env.
+   */
+  void clearRewriterCaches();
```

Comment:
This method is extremely rare (in fact, it should probably just be deleted altogether from Rewriter).  Regardless, to keep the interface on EnvObj small, let's have the caller of this method access the rewriter manually.